### PR TITLE
Switch to device_frame

### LINKED ISSUE #56: Preview on Landscape Mode
Some users wished to preview their widgets not only on devices with portrait mode but also on landscape mode.

### LINKED ISSUE #66: MediaQuery data not specified for device.
Widgets that depend on `MediaQuery` data such as `MediaQuery.of(context).size`  will receive data from the Widgetbook window size instead of the device size as shown bellow.

https://user-images.githubusercontent.com/16095735/143028314-568064aa-6c85-499c-91c8-1a3329d9333e.mov

### PR DESCRIPTION
**Pull request by edTheGuy00**
This pull request updates WidgetBook to use [device_frame](https://github.com/aloisdeniel/flutter_device_preview/tree/master/device_frame) instead of the built in `widgetbook_models`. 

The main benefits of using `device_frame` are:

- A List of Ready to use Device Frame UI
- Correct MediaQuery data for all devices. (fixes #66)
- User toggable frame
- Built in Virtual Keyboard
- Orientation Switching (fixes #56 )

This changes renders [widgetbook_models](https://github.com/widgetbook/widgetbook/tree/main/packages/widgetbook_models) package completely useless. The only braking change for users is switching from [device.dart](https://github.com/widgetbook/widgetbook/blob/main/packages/widgetbook_models/lib/src/devices/device.dart) to [devices.dart](https://github.com/aloisdeniel/flutter_device_preview/blob/master/device_frame/lib/src/devices/devices.dart) from the `device_frame` library. 

`devices.dart` will be [exposed from widgetbook](https://github.com/edTheGuy00/widgetbook/blob/0b4e4e661bd73584bead61d159279c98f4aecff5/packages/widgetbook/lib/widgetbook.dart#L1) so users do not need to add this dependency separately. The only downside is that `device_frame` does not include watch devices which are included in widgetbook, But I could make a pull request to `device_frame` later on to add more devices into it. 

TODO:

- [ ] Update Documentation
- [ ] Clean Up Code
- [ ] Update Tests

## Preview

Screen Shot 2021-11-28 at 13 41 03

Screen Shot 2021-11-28 at 13 41 49

Screen Shot 2021-11-28 at 13 41 32

### CODE REVIEW

**Comment by JensHor** [comment]
@edTheGuy00 this already looks amazing 👍🏻. I also just realised that the generator has to be adjusted as well since devices was switched to `device_frame`. Are you planning to make these changes? otherwise i can make the adjustments as well.

**Comment by JensHor** [comment]
making the adjustments might be a problem because the build_runner package does not work too well with flutter dependencies and `device_frame` unfortunately depends on flutter. A solution would be to either map `widgetbook` devices to `device_frame` devices or create a new device_frame package that just holds the models (however that has to be discussed with the device_frame maintainers).

**Comment by edTheGuy00** [comment]
@JensHor Hey, sorry for the delay. Kinda been busy with work, we have just been using my branch of widgetbook for the time being :sweat_smile:

I think having a separate package that we collaboratively maintain may be the best approach. I'll open an issue on [Device Preview](https://github.com/aloisdeniel/flutter_device_preview) to discus with the maintainers about this. Otherwise, mapping the devices will be the secondary option.

**Comment by edTheGuy00** [comment]
> @JensHor Hey, sorry for the delay. Kinda been busy with work, we have just been using my branch of widgetbook for the time being sweat_smile
> 
> I think having a separate package that we collaboratively maintain may be the best approach. I'll open an issue on [Device Preview](https://github.com/aloisdeniel/flutter_device_preview) to discus with the maintainers about this. Otherwise, mapping the devices will be the secondary option.

I took a quick look at the `device_frame` code. I think it will be more of a hassle on their side to branch off into a separate package than us to simple map the devices in one `models` class.

**Comment by JensHor** [comment]
Alright. Mapping the devices seems to be a good workaround and should also would simplify `widgetbook_generator` development 👍🏻

**Comment by JensHor** [comment]
hey @edTheGuy00. I really love the changes you've made 💙. Are you planning to complete the implementation? Otherwise, i'd make plans to complete this myself. Let me know!

**Comment by leonardosilva-asaas** [comment]
@JensHor I was going to ask about updates but then I saw your comment is from 6 hours ago and already got my response. Please do it! 

Since I started using WidgetBook I noticed this was something that was missing. I have used `device_preview` before and they use `device_frame` as well. It is awesome. I think it will be a great addition and will make the WidgetBook much better.

**Comment by edTheGuy00** [comment]
> hey @edTheGuy00. I really love the changes you've made blue_heart. Are you planning to complete the implementation? Otherwise, i'd make plans to complete this myself. Let me know!

@JensHor Yes, I've been meaning to get back to it but I haven't had the time do to it these past few weeks, so sorry to leave you hanging. Please feel free to complete it and I might have some free time mid next month if it still hasn't been implemented by then. 😊

File: packages/widgetbook/lib/src/widgets/device_bar.dart [hunk 1]
```diff
@@ -56,6 +56,77 @@ class DeviceBar extends StatelessWidget {
             color: context.theme.hintColor,
           ),
         ),
+        const SizedBox(
+          width: 40,
+        ),
+        Tooltip(
```
Comment (tieorange) [inline packages/widgetbook/lib/src/widgets/device_bar.dart:62]: Extract widget please

File: packages/widgetbook/lib/src/widgets/device_bar.dart [hunk 2]
```diff
@@ -56,6 +56,77 @@ class DeviceBar extends StatelessWidget {
             color: context.theme.hintColor,
           ),
         ),
+        const SizedBox(
+          width: 40,
```
Comment (tieorange) [inline packages/widgetbook/lib/src/widgets/device_bar.dart:60]: Can be oneliner (without `,`)

File: packages/widgetbook/lib/src/widgets/device_item.dart [hunk 3]
```diff
@@ -35,24 +35,24 @@ class DeviceItem extends StatelessWidget {
...
-        return Icon(
-          Icons.watch,
-          color: color,
-        );
-      case DeviceType.mobile:
-        return Icon(
-          Icons.smartphone,
-          color: color,
-        );
+      // case DeviceType.watch:
```
Comment (tieorange) [inline packages/widgetbook/lib/src/widgets/device_item.dart:46]: remove commented code or add a TODO

**Pull request closed without merging**